src/lseq.rs

```diff
@@ -0,0 +1,254 @@
+//! # LSEQ
+//!
+//! An LSEQ tree is a CRDT for storing sequences of data (Strings, ordered lists).
+//! It provides an efficient view of the stored sequence, with fast index, insertion and deletion
+//! operations.
+//!
+//! LSEQ [1] is a member of the LOGOOT [2] family of algorithms for CRDT sequences. The major difference
+//! with LOGOOT is in the _allocation strategy_ that LSEQ employs to handle insertions.
+//!
+//! Internally, LSEQ views the sequence as the leaves of an ordered, exponential tree. An
```

Comment:
TODO: make sure that the values are actually not at the leaves

Reply:
?

Reply:
Ah sorry, was on a call while looking at this code, didn't have time to expand.

As I understand it, values are stored at the end of an identifier, but that identifier is not necessarily pointing at a leaf. It could be an internal node of the tree.

Reply:
Yeah I think that's right, `e1 = Entry { id: vec![(0, "a")], val: V1 ... }` and `e2 = Entry { id: vec![(0, "a"), (1, "b")], val: V2 ... }` Are both valid with `e1` appearing before `e2` since `e1.id < e2.id`.

```suggestion
//! Internally, LSEQ views the sequence as the nodes of an ordered, exponential tree. An
```

Reply:
> As I understand it, values are stored at the end of an identifier, but that identifier is not necessarily pointing at a leaf. It could be an internal node of the tree. 

Yep! You're right, my mistake.